### PR TITLE
improve placement pass tests code coverage

### PR DESCRIPTION
**Pull request by wojtuss**
Test classes are introduced to test private placement passes' methods. No changes in the existing methods.

test=develop

### CODE REVIEW

**Review by Sand3r- — APPROVED** [review]
LGTM

File: paddle/fluid/framework/ir/mkldnn/mkldnn_placement_pass_tester.cc [hunk 1]
```diff
@@ -21,112 +21,129 @@ namespace paddle {
...
+                unsigned expected_use_mkldnn_true_count) {
+    auto prog = BuildProgramDesc();
 
-void MainTest(std::initializer_list<std::string> mkldnn_enabled_op_types,
-              unsigned expected_use_mkldnn_true_count) {
-  auto prog = BuildProgramDesc();
+    std::unique_ptr<ir::Graph> graph(new ir::Graph(prog));
 
-  std::unique_ptr<ir::Graph> graph(new ir::Graph(prog));
+    auto pass = PassRegistry::Instance().Get("mkldnn_placement_pass");
```
Comment (lidanqing-vv) [inline paddle/fluid/framework/ir/mkldnn/mkldnn_placement_pass_tester.cc:101]: LGTM.   I just have one doubt, after we get the mkldnn pass here, do we expect its GetPlaceName() is "MKLDNN" ?

**Review by lidanqing-vv — COMMENTED** [review]
LGTM

File: paddle/fluid/framework/ir/mkldnn/mkldnn_placement_pass_tester.cc (hunk 1 above)
Comment (wojtuss) [reply paddle/fluid/framework/ir/mkldnn/mkldnn_placement_pass_tester.cc:101]: Yes. The `MKLDNNPlacementPass` and `CUDNNPlacementPass` classes override the `PlacementPassBasebase` class' method `GetPlacementName()`. They return `MKLDNN` and `cuDNN`, respectively.

**Review by luotao1 — APPROVED** [review]

**Pull request merged**